### PR TITLE
Fix pre-existing TracingTest lifecycle assertion failures (BT-2116)

### PR DESCRIPTION
**Pull request by jamesc**
## Summary

Make telemetry attachment deterministic so the 4 lifecycle-event tests in `stdlib/test/tracing_test.bt` no longer rely on opportunistic startup.

Linear: https://linear.app/beamtalk/issue/BT-2116

## What changed

- **`runtime/apps/beamtalk_runtime/src/beamtalk_runtime.app.src`**: Declare `telemetry` and `telemetry_poller` in the application's `applications` list. OTP now starts them before `beamtalk_trace_store`'s `init/1` runs.
- **`runtime/apps/beamtalk_runtime/src/beamtalk_trace_store.erl`**: Replace the conditional `case application:ensure_all_started(telemetry) of {ok, _} -> attach; {error, _} -> log` block with unconditional `attach_telemetry_handlers/0` + `configure_poller/0` calls. Drop the redundant `ensure_all_started(telemetry_poller)` from `configure_poller/0`.
- **`stdlib/test/tracing_test.bt`**: Fix the stale comment header that claimed "telemetry is not available in BUnit" — it is. Lifecycle events from compiled actors are captured normally.
- **`crates/beamtalk-examples/corpus.json`**: Regenerated (picks up the comment update).

## Why

The trace store previously called `application:ensure_all_started(telemetry)` opportunistically, with a comment about avoiding "environments where telemetry isn't on the code path." Verified via diagnostic: telemetry is on the BUnit code path (it's a hex dep in `runtime/rebar.config`, and `beam_pa_args` adds its `ebin/` to the BEAM path). Whenever the conditional landed in the `{error, _}` branch — e.g. due to a startup race — handlers were silently skipped and the lifecycle tests failed mysteriously.

Declaring the dependency in `app.src` lets OTP guarantee startup ordering, eliminating the race.

## Test plan

- [x] `just build` clean
- [x] `just test-bunit` × 5 consecutive runs, all 1904 tests pass
- [x] `just test-bunit stdlib/test/tracing_test.bt` × 10 consecutive runs (35/35 each)
- [x] `just test-runtime` (3566 + 1242 EUnit tests, 0 failures)
- [x] `just ci` green end-to-end

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Clarified that telemetry monitoring is available in the testing harness and adjusted test comments about expected trace contents.

* **Chores**
  * Runtime now starts telemetry services automatically.
  * Simplified initialization to assume telemetry availability, with related poller setup streamlined.

### CODE REVIEW

File: stdlib/test/tracing_test.bt [hunk 1]
```diff
@@ -5,9 +5,12 @@
...
 //
-// Note: telemetry is not available in BUnit, so dispatch-based aggregates
-// and trace events are not auto-populated. We test the API surface, error
-// behavior, and live health rather than trace data content.
+// Telemetry is available in the BUnit harness (declared as a dependency of
+// beamtalk_runtime, so OTP starts it before the trace store gen_server's init/1
+// runs). Lifecycle events from compiled actors are therefore captured here.
+// Method-dispatch telemetry is opt-in per actor and not exercised by these
+// tests — the API-surface tests still assert empty results when no dispatches
+// have been recorded.
```
[the remark this answers is not in the document]
Comment (jamesc) [reply stdlib/test/tracing_test.bt]: Thanks — you're right, the "opt-in per actor" framing was inaccurate. Actor dispatch is always instrumented via `telemetry:span/3` when telemetry is loaded; there's no per-actor toggle. Rephrased the comment in 73f3bba9 to describe the actual condition: dispatch traces are empty here because no method dispatches are performed against an actor while tracing is enabled, not because of an opt-in.

File: runtime/apps/beamtalk_runtime/src/beamtalk_trace_store.erl [hunk 2]
```diff
@@ -432,18 +432,12 @@ init([]) ->
...
-        {error, _} ->
-            ?LOG_INFO("telemetry not available, skipping handler attachment", #{}),
-            ok
-    end,
+    %% telemetry and telemetry_poller are declared in beamtalk_runtime.app.src,
+    %% so OTP guarantees they have started by the time this init/1 runs.
+    %% Attach handlers unconditionally — silent attachment failures previously
+    %% manifested as flaky lifecycle-event tests (BT-2116).
+    attach_telemetry_handlers(),
+    configure_poller(),
```
[the remark this answers is not in the document]
Comment (jamesc) [reply runtime/apps/beamtalk_runtime/src/beamtalk_trace_store.erl]: Good catch — the runtime had `maybe_span/3` and `maybe_execute_telemetry/3` in `beamtalk_actor.erl` (lines 1163, 1178) that gate on `erlang:function_exported(telemetry, ...)`, so removing the guard here introduced an inconsistency. Restored a `function_exported(telemetry, attach, 4)` guard around `attach_telemetry_handlers/0` and `configure_poller/0` in 73f3bba9, mirroring that pattern. The `app.src` declaration still ensures telemetry starts in normal flow; the guard handles edge cases like direct `gen_server:start_link/3` in a partial test harness.
